### PR TITLE
run-checker: add more builds with FIPS enabled and other disabled.

### PR DESCRIPTION
**Pull request by paulidale**
Not all of these work yet, this won't be merged until they do.

### CODE REVIEW

**Comment by paulidale** [comment]
The FIPS no-des fix is in #15105, the rest seem good locally.

**Review by mattcaswell — APPROVED** [review]

**Comment by paulidale** [comment]
Added a fixup to remove the duplicate entry.  Reapproval required.

File: run-checker/run-checker.sh [hunk 1]
```diff
@@ -46,7 +46,13 @@ no-tls1_2-method no-dtls1-method no-dtls1_2-method no-siphash no-tls1_3 no-sm2
 no-sm3 no-sm4 enable-trace no-legacy no-cached-fetch no-autoload-config
 'no-buildtest-c++' no-bulk no-cmp no-ktls no-module no-padlockeng
 no-pinshared no-secure-memory no-siv no-uplink enable-acvp-tests enable-fips
-'enable-fips no-fips-securitychecks'
+'enable-fips no-fips-securitychecks' 'enable-fips enable-acvp-tests'
+'enable-fips no-cmac' 'enable-fips no-cmac' 'enable-fips no-des'
```
Comment (paulidale) [inline run-checker/run-checker.sh:50]: On second thoughts are the FIPS + disabled algorithms actually meaningful?   It seems unlikely that the FIPS provider could be built in an approved manner if they are excluded.

Comment (t8m) [reply run-checker/run-checker.sh:50]: Well, somebody could do a private validation with FIPS and disabled algorithms. But I am not sure we need to worry about such in runchecker builds.

Comment (paulidale) [reply run-checker/run-checker.sh]: I've pulled the FIPS + disabled algorithms options.

**Review by t8m — APPROVED** [review]

**Comment by paulidale** [comment]
Merged to master.  Let the fun begin.
Thanks for all of the helpful comments.

**Pull request closed without merging**